beacon-chain/blockchain/process_block.go

```diff
@@ -529,11 +533,43 @@ func (s *Service) runLateBlockTasks() {
 	}
 }
 
+// missingIndices uses the expected commitments from the block to determine
+// which BlobSidecar indices would need to be in the database for DA success.
+// It returns a map where each key represents a missing BlobSidecar index.
+// An empty map means we have all indices; a non-empty map can be used to compare incoming
+// BlobSidecars against the set of known missing sidecars.
+func missingIndices(bs *filesystem.BlobStorage, root [32]byte, expected [][]byte) (map[uint64]struct{}, error) {
```

Comment:
Is there a test for this one?

Reply:
added https://github.com/prysmaticlabs/prysm/pull/13190/commits/121ef9052eb983088e09ec2efeeae7ee322d9ad0